catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -331,31 +319,41 @@ public UpdateResponse update(UpdateRequest updateRequest) throws IngestException
             .map(Serializable::toString)
             .collect(Collectors.toList());
 
-    /* 1a. Create the old Metacard Query */
-    String attributeQuery = getQuery(attributeName, identifiers);
-
-    SolrQuery query = new SolrQuery(attributeQuery);
-    // Set number of rows to the result size + 1.  The default row size in Solr is 10, so this
-    // needs to be set in situations where the number of metacards to update is > 10.  Since there
-    // could be more results in the query response than the number of metacards in the update
-    // request,
-    // 1 is added to the row size, so we can still determine whether we found more metacards than
-    // updated metacards provided
-    query.setRows(updates.size() + 1);
-    QueryResponse idResults = null;
+    final Map<Serializable, Metacard> idToMetacardMap = new HashMap<>();
+    if (Metacard.ID.equals(attributeName)) {
+      try {
+        idToMetacardMap.putAll(
+            client
+                .getIds(new HashSet<>(identifiers))
```

Comment:
❓ Is there an actual need for cloning the `identifiers` set or would just wrapping it in an unmodifiable collection do the trick?

Reply:
Well they were previously created in a List within this same scope - so we shouldn't need to protect them since we're using them to generate yet another new object within this same scope. Maybe he's doing it for removing duplicates? but that could also be achieved by doing a `.distinct()` call on the stream (and prevent the allocation). Not sure 🤔

Reply:
`identifiers` is a list and `getIds` takes a set.  I can change the other uses of `identifiers` to use a set instead.

Reply:
I think with the latest code the wrapping in HashSet<> can be removed as `identifiers` is already a set.